docs/howto/how-to-make-tensorflow-plugin-with-ui/tensorflow-plugin-with-ui.md

```diff
@@ -0,0 +1,1488 @@
+
+  useEffect(() => {
+    handleUpdateModel(model)
+  }, []);
+
+  return (
+    <Box p={4}>
+     <Grid container spacing={4} direction="column" alignItems="center">
+        <Grid item>
+          <Alteryx className={classes.alteryx} />
+        </Grid>
+        <Grid item>
+          <Typography variant="h3">
+            To get started, edit src/index.tsx
+          </Typography>
+        </Grid>
+      </Grid>
+    </Box>
+  )
+}
+
+const Tool = () => {
+  return (
+    <DesignerApi messages={{}}>
+      <AyxAppWrapper> 
+        <App />
+      </AyxAppWrapper>
+    </DesignerApi>
+  )
+}
+
+ReactDOM.render(
+  <Tool />,
+  document.getElementById('app')
+);
+```
+
+#### 1.2.5.2. backend/ayx_plugins/text_classifier.py
+```python
+class TextClassifier(PluginV2):
+    """A sample Plugin that passes data from an input connection to an output connection."""
+
+    def __init__(self, provider: AMPProviderV2):
+        """Construct the plugin."""
+        self.name = "TextClassifier"
+        self.provider = provider
+        self.provider.io.info(f"{self.name} tool started")
+
+    def on_record_batch(self, batch: "pa.Table", anchor: Anchor) -> None:
+        # truncated code
+
+    def on_incoming_connection_complete(self, anchor: Anchor) -> None:
+        # truncated code
+
+    def on_complete(self) -> None:
+        # truncated code
+```
+
+## 1.3. Backend: A Naive Tensorflow Text Classifier
+
+Here, we use the term naive to describe a tool that assumes perfect input in a known structure. 
+Later we will add input fields that accept arbitrary values. But we can start with controlled, known values first to simplify testing.
+To further simplify the transition to User populated or arbitrary data, we can also create a constants.py file.
+For now, leave it blank. We'll need it later to create static values for our tests.
+
+We update our tool by "mode" (recall our modes are: `DATA, PREVIEW, TRAIN, PREDICT`).
+Considering we need to test building the model in that order anyway, this will make reasoning about our code more straightforward.
+
+
+### 1.3.1. Data Mode
+
+For this guide, we'll be using the [imdb movie review dataset](https://ai.stanford.edu/~amaas/data/sentiment/aclImdb_v1.tar.gz).
+This archive is a dataset of movie reviews and their labels for classification.
+The labels are binary positive or negative reviews.
+The reviews are scrapped from the IMDb website and then appropriately labeled.
+You may download it now or during the plugin runtime.
+Here we download it and reference it locally.
+Note that we get this in `.tar.gz` format, so you need to decompress this before using the data.
+Once you decompress, note the location as a constant in the `constants.py` module we created earlier.
+Something like this will do:
+
+`DATA_SRC = '/path/to/archive/aclImdb_v1.tar/'`
+
+Next, open up `backend/ayx_plugins/text_classifier.py`. It's finally time to add some functional code!
+
+Update the `__init__` method to the below:
```

Comment:
TW review:

Next, open `backend/ayx_plugins/text_classifier.py`. It's finally time to add some functional code!

Update the `__init__` method to this: